amundsen_application/static/js/components/SearchPage/index.tsx

```diff
@@ -96,7 +98,10 @@ export class SearchPage extends React.Component<SearchPageProps, SearchPageState
       const { searchTerm, pageIndex, selectedTab } = params;
       const { term, index, currentTab } = this.getSanitizedUrlParams(searchTerm, pageIndex, selectedTab);
       this.setState({ selectedTab: currentTab });
-      this.props.searchAll(term, this.createSearchOptions(index, currentTab));
+      const prevTerm = qs.parse(prevProps.location.search).searchTerm;
```

Comment:
Rather than re-parsing the previous props location, you can pull in the prev state and use that value. Interface for this function is:

`componentDidUpdate(prevProps, prevState, snapshot)`

Reply:
oh right, I could just set it prevProps.searchTerm right?

Reply:
There's an important distinction between `state` and `props`. With the way we're using `Redux/Saga` the 'container' pulls values from the `globalState` and sends it as `props` on the `component`. 

Sometimes the component can use the props as-is, but other times it needs to consume props to update its internal `state`.